lisa/src/main/java/it/unive/lisa/analysis/impl/heap/pointbased/AllocationSite.java

```diff
@@ -20,61 +21,70 @@
+	public AllocationSite(ExternalSet<Type> types, String id, boolean isWeak, CodeLocation location) {
+		this(types, id, null, isWeak, location);
 	}
 
 	/**
 	 * Builds a strong allocation site from its source code location and its
 	 * field.
 	 * 
-	 * @param types the runtime types of this allocation site
-	 * @param id    the source code location string representation where this
-	 *                  allocation site has been allocated
-	 * @param field the field of this allocation site
+	 * @param types    the runtime types of this allocation site
+	 * @param id       the source code location string representation where this
+	 *                     allocation site has been allocated
+	 * @param field    the field of this allocation site
+	 * @param location the code location of the statement that has generated
+	 *                     this expression
 	 */
-	public AllocationSite(ExternalSet<Type> types, String id, SymbolicExpression field) {
-		this(types, id, field, false);
+	public AllocationSite(ExternalSet<Type> types, String id, SymbolicExpression field, CodeLocation location) {
+		this(types, id, field, false, location);
 	}
 
 	/**
 	 * Builds an allocation site from its source code location and its field and
 	 * specifying if it is weak.
 	 * 
-	 * @param types  the runtime types of this allocation site
-	 * @param id     the source code location string representation where this
-	 *                   allocation site has been allocated
-	 * @param field  the field of this allocation site
-	 * @param isWeak boolean value specifying if this allocation site is weak
+	 * @param types    the runtime types of this allocation site
+	 * @param id       the source code location string representation where this
+	 *                     allocation site has been allocated
+	 * @param field    the field of this allocation site
+	 * @param isWeak   boolean value specifying if this allocation site is weak
+	 * @param location the code location of the statement that has generated
+	 *                     this expression
 	 */
-	public AllocationSite(ExternalSet<Type> types, String id, SymbolicExpression field, boolean isWeak) {
-		super(types, "pp@" + id + (field == null ? "" : "[" + field + "]"), isWeak);
+	public AllocationSite(ExternalSet<Type> types, String id, SymbolicExpression field, boolean isWeak,
+			CodeLocation location) {
+		super(types, "pp@" + id + (field == null ? "" : "[" + field + "]"), isWeak, location);
 		this.id = id;
 	}
 
 	/**
-	 * Returns the source code location string representation where this
-	 * allocation site has been allocated.
+	 * Returns the code location string representation where this allocation
+	 * site has been allocated.
 	 * 
-	 * @return the source code location string representation where this
-	 *             allocation site has been allocated.
+	 * @return the code location string representation where this allocation
+	 *             site has been allocated.
 	 */
 	public String getId() {
```

Comment:
Maybe a more accurate name of for the field/method would be better, as `id` does not explicitly hint the code location